legend-engine-xts-data-space/legend-engine-xt-data-space-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/DataSpaceCompilerExtension.java

```diff
@@ -282,6 +310,71 @@ else if (dataSpace.supportInfo instanceof DataSpaceSupportCombinedInfo)
                         }
                         metamodel._supportInfo(supportInfo);
                     }
+                },
+                (dataSpace, context) ->
+                {
+                    // Move checking mapping/runtime utilized in the function within the curated template query does not align with the mapping/runtime applied in the execution context
+                    // to the sixth pass because function body and parameter are processed in the fifthPassBuilder
+                    // if we do the checking in the fifth pass, function body and parameter will be an empty list
+                    if (dataSpace.executables != null)
+                    {
+                        dataSpace.executables.forEach(executable ->
+                        {
+                            if (executable instanceof DataSpaceTemplateExecutable)
+                            {
+                                if (((DataSpaceTemplateExecutable) executable).query instanceof PackageableElementPointer)
+                                {
+                                    try
+                                    {
+                                        FunctionDefinition<?> templateExecutableQuery = (FunctionDefinition<?>) context.resolvePackageableElement(FunctionDescriptor.functionDescriptorToId(((PackageableElementPointer) ((DataSpaceTemplateExecutable) executable).query).path), ((PackageableElementPointer) ((DataSpaceTemplateExecutable) executable).query).sourceInformation);
+                                        if (templateExecutableQuery instanceof Root_meta_pure_metamodel_function_ConcreteFunctionDefinition_Impl)
+                                        {
+                                            Optional<? extends ValueSpecification> fromFunc = ((Root_meta_pure_metamodel_function_ConcreteFunctionDefinition_Impl<?>) templateExecutableQuery)._expressionSequence().toList().stream()
+                                                    .filter(func -> func instanceof Root_meta_pure_metamodel_valuespecification_SimpleFunctionExpression_Impl && ((Root_meta_pure_metamodel_valuespecification_SimpleFunctionExpression_Impl) func)._functionName.equals("from")).findAny();
+                                            if (fromFunc.isPresent())
+                                            {
+                                                Root_meta_pure_metamodel_valuespecification_SimpleFunctionExpression_Impl fromFuncExpression = (Root_meta_pure_metamodel_valuespecification_SimpleFunctionExpression_Impl) fromFunc.get();
+
```

Comment:
Can we look into using pure built in functions for these types of checks to match things like function tests, so if we need to improve the logic of extracting runtime/mapping we can do it in one place ? 
See:
`Root_meta_pure_functions_meta_extractRuntimesFromFunctionDefinition_FunctionDefinition_1__Runtime_MANY_`